core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/scatterplot/ScatterplotOpExec.java

```diff
@@ -15,30 +16,20 @@
  */
 public class ScatterplotOpExec extends MapOpExec {
     private final ScatterplotOpDesc opDesc;
+    private final OperatorSchemaInfo operatorSchemaInfo;
 
-    public ScatterplotOpExec(ScatterplotOpDesc opDesc) {
+    public ScatterplotOpExec(ScatterplotOpDesc opDesc, OperatorSchemaInfo operatorSchemaInfo) {
         this.opDesc = opDesc;
+        this.operatorSchemaInfo = operatorSchemaInfo;
         this.setMapFunc((Function1<Tuple, Tuple> & Serializable) this::processTuple);
     }
 
     public Tuple processTuple(Tuple t) {
-        List<Object> resultObjects = new ArrayList<>();
+        Object[] resultObjects = new Object[2];
 
-        Schema resultSchema;
-        if(opDesc.isGeometric)
-            resultSchema = Schema.newBuilder().add(
-                    new Attribute("xColumn", t.getSchema().getAttribute(opDesc.xColumn).getType()),
-                    new Attribute("yColumn", t.getSchema().getAttribute(opDesc.yColumn).getType())
-            ).build();
+        resultObjects[0] = t.getField(opDesc.xColumn);
```

Comment:
Please check ScatterPlot changes properly. It seems like you are changing something not related to just schema changes. Did you merge the current master correctly?

Reply:
Same comment as [here](https://github.com/Texera/texera/pull/1166#discussion_r655019089). I moved the code to make it work correctly. @sadeemsaleh to confirm.

Reply:
@sadeemsaleh has confirmed the change https://github.com/Texera/texera/pull/1166#discussion_r659358110